tests/tests_integration/test_configs.py

```diff
@@ -77,6 +81,90 @@ def test_bad_paths(self, project, bad_pattern, path_type):
 
         assert "must contain the full folder path with no " in str(e.value)
 
+    @pytest.mark.parametrize("path_type", ["local_path", "central_path"])
+    def test_non_existant_local_path(
+        self, no_cfg_project, tmp_path, path_type
+    ):
+        """
+        Check that if the `local_path` and `central_path` that holds the
+        project root does not exist when passed to configs, that an error
+        is raised. Note that this error is only raised for `central_path`
+        if `connection_method` is `"local_filesystem"`.
+        See `test_additional_error_text_when_ssh_used()` for when ssh is used.
+        """
+        (
+            non_existant_path,
+            existant_path,
+        ) = self.get_exists_and_does_not_exist_path(tmp_path)
+
+        if path_type == "local_path":
+            local_path = non_existant_path
+            central_path = existant_path
+        else:
+            local_path = existant_path
+            central_path = non_existant_path
+
+        with pytest.raises(BaseException) as e:
+            no_cfg_project.make_config_file(
+                local_path / no_cfg_project.project_name,
+                central_path / no_cfg_project.project_name,
+                "local_filesystem",
+            )
+
+        assert f"The {path_type}: {non_existant_path} that the project" in str(
+            e.value
+        )
+
+    def test_additional_error_text_when_ssh_used(
+        self, no_cfg_project, tmp_path
+    ):
+        """
+        If SSH is used as `connection_method`, if `local_path` does not exist
+        an extra message is printed to warn to check the `central_path`, because
+        it cannot be checked.
+
+        Currently if SSH is used and the central path does not exist,
+        no error is raised because it is not possible to check.
+        """
+        (
+            non_existant_path,
+            existant_path,
+        ) = self.get_exists_and_does_not_exist_path(tmp_path)
+
+        with pytest.raises(BaseException) as e:
+            no_cfg_project.make_config_file(
+                non_existant_path / no_cfg_project.project_name,
+                existant_path / no_cfg_project.project_name,
+                "ssh",
+                central_host_id="fake_id",
+                central_host_username="fake_username",
+            )
+
+        assert (
+            "Also make sure the central_path`, is correct, as datashuttle "
+            "cannot check it via SSH at this stage." in str(e.value)
+        )
+
+        # This should not raise an error, even though the path does not
+        # exist, because it is not possible to check over SSH.
```

Comment:
FYI you cand do:
```python
from contextlib import nullcontext as does_not_raise

with does_not_raise():
```

I don't know if it's useeful here, but it allows you to parametrize arguments against multiple expected exceptions, one of which could be `does_not_raise()` (in cases where some values raise exceptions and others don't)

Reply:
This is really cool, in this specific case I will leave it because the order of the arguments is swapped rather than parameterising a single argument so it might be a little fiddly. However will definitely remember this for future, thanks!